### DISCUSSION TITLE
Convert Transform4f matrix to Numpy matrix or Python list

### DISCUSSION THREAD

**Discussion by HeyLongHoang** [mitsuba-renderer/mitsuba3#1116] [Q&A] [answered]
Hi all, I am currently having difficulty converting Transform4 matrix to Numpy array or general Python list. I tried:
np.array(mi.traverse(sensor)['to_world']). However, this gives back a matrix of shape () and doesn't allow indexing. Do you know how to fix this? Thank you!
<img width="1440" alt="Screenshot 2024-03-22 at 16 23 48" src="https://github.com/mitsuba-renderer/mitsuba3/assets/79366791/89a62c64-9f1e-41b3-9a73-6d8a3ef6b1c4">

**Comment by HeyLongHoang** [comment]
I found the solution. You need to call M.matrix and then wrap it as Numpy array.

**Reply by njroussel** [reply, marked as answer]
Indeed, the `Transform` class is a convenience wrapper to handle transformations of points, and normals differently. The raw underyling matrix is what you most likely want here, and you can access it with `my_transform.matrix`.
